pkg/identity/cache/allocator.go

```diff
@@ -443,7 +443,7 @@ func (m *CachingIdentityAllocator) WatchRemoteIdentities(backend kvstore.Backend
 		return nil, fmt.Errorf("Error setting up remote allocator backend: %s", err)
 	}
 
-	remoteAlloc, err := allocator.NewAllocator(GlobalIdentity{}, remoteAllocatorBackend)
+	remoteAlloc, err := allocator.NewAllocator(GlobalIdentity{}, remoteAllocatorBackend, allocator.WithEvents(m.IdentityAllocator.GetEvents()))
```

Comment:
One problem with this is that Allocator.Delete closes the events channel. It is only called in tests right now but it would cause a panic. We could have a pipe that uses a new channel, and would be notified when `.events` is closed